### PR TITLE
Trip service facade

### PR DESCRIPTION
**Pull request by kotycheese**

### CODE REVIEW

File: Travel-Agency-api/src/main/java/fi/muni/cz/pa165/travelagency/dto/TripCreateDTO.java [hunk 1]
```diff
@@ -1,7 +1,175 @@
...
+    private Date dateTo;
+
+    @NotNull
+    private String destination;
+
+    @Min(0)
+    private int availableSpots;
+
+    @NotNull
+    private Long excursionId;
```
Comment (445611) [inline Travel-Agency-api/src/main/java/fi/muni/cz/pa165/travelagency/dto/TripCreateDTO.java]: entity Excrusion has collection of Excursion ,this way when you are creating you can set only one excursion, is that intended ?

File: Travel-Agency-service/src/main/java/fi/muni/cz/pa165/travelagency/service/facade/TripFacadeImpl.java [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    private ExcursionService excursionService;
+
+    @Autowired
+    private BeanMappingService beanMappingService;
+
+    @Override
+    public Long createTrip(TripCreateDTO trip) {
+        Trip mapped = beanMappingService.mapTo(trip, Trip.class);
+
+        if(trip.getExcursionId() != null) {
```
Comment (445611) [inline Travel-Agency-service/src/main/java/fi/muni/cz/pa165/travelagency/service/facade/TripFacadeImpl.java]: mapTo() above will not map ExcursionID into Excursion. so you anyway have to set it, dont need to check it I think.

File: Travel-Agency-api/src/main/java/fi/muni/cz/pa165/travelagency/dto/TripCreateDTO.java (hunk 1 above)
Comment (kotycheese) [reply Travel-Agency-api/src/main/java/fi/muni/cz/pa165/travelagency/dto/TripCreateDTO.java]: yes, it is written in the same way as example e-shop project, i would call it an "initial excursion", dont know if there could be some problems with the whole collection..

File: Travel-Agency-service/src/main/java/fi/muni/cz/pa165/travelagency/service/facade/TripFacadeImpl.java (hunk 2 above)
Comment (kotycheese) [reply Travel-Agency-service/src/main/java/fi/muni/cz/pa165/travelagency/service/facade/TripFacadeImpl.java]: not sure i understand.. that "if" is for adding the initial excursion to the set of excursions in mapped trip, if the id is null, the set will remain empty

**Pull request merged**